ServerCore/FileManager.cs

```diff
@@ -138,7 +138,7 @@ public static async Task DeleteBlobAsync(Uri fileUri)
         /// </summary>
         /// <param name="eventId">Event the directory should be associated with</param>
         /// <param name="puzzleDirectoryName">Name of a subdirectory to concatenate. Use empty string for none.</param>
-        private static async Task<CloudBlobDirectory> GetPuzzleDirectoryAsync(int eventId, string puzzleDirectoryName)
+        public static async Task<CloudBlobDirectory> GetPuzzleDirectoryAsync(int eventId, string puzzleDirectoryName)
```

Comment:
OK I know that I suggested that you make this function public, but I forgot that it would leak CloudBlobDirectory out of this file. I think it's better to keep all blob know-how inside FileManager; can you please change this back to private and make a new public method that returns the (absolute) Uri, which is all you care about anyway?